src/components/cylc/gscan/GScan.vue

```diff
@@ -102,79 +96,45 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
         </v-menu>
       </div>
       <!-- data -->
-      <div
-        v-if="!isLoading"
-        class="c-gscan-workflows flex-grow-1"
-      >
-        <tree
-          :filterable="false"
-          :expand-collapse-toggle="false"
-          :workflows="workflows"
-          :stopOn="['workflow']"
-          :autoExpandTypes="['workflow-part', 'workflow']"
-          class="c-gscan-workflow ma-0 pa-0"
-          ref="tree"
-        >
+      <div v-if="!isLoading" class="c-gscan-workflows flex-grow-1">
+        <tree :filterable="false" :expand-collapse-toggle="false" :workflows="workflows" :stopOn="['workflow']" :autoExpandTypes="['workflow-part', 'workflow']" class="c-gscan-workflow ma-0 pa-0" ref="tree">
+          <!-- this overwrites the node slot in the tree-item template, these are ALWAYS inserted where the node slot is regardless of its current value -->
           <template v-slot:node="scope">
-            <workflow-icon
-              class="mr-2"
-              v-if="scope.node.type === 'workflow'"
-              :status="scope.node.node.status"
-              v-cylc-object="scope.node"
-            />
-            <v-list-item
-              :to="workflowLink(scope.node)"
-            >
+
+            <workflow-icon class="mr-2" v-if="scope.node.type === 'workflow' || !checkForBranchingLineage(scope.node)" :status="getLastDescendent(scope.node).node.status || 'waiting'" v-cylc-object="getLastDescendent(scope.node)" />
+
+            <v-list-item :to="workflowLink(scope.node)">
               <v-list-item-title>
                 <v-layout align-center align-content-center d-flex flex-nowrap>
-                  <v-flex
-                    v-if="scope.node.type === 'workflow-part'"
-                    class="c-gscan-workflow-name"
-                  >
-                    <span>{{ scope.node.name || scope.node.id }}</span>
+
+                  <v-flex v-if="scope.node.type === 'workflow-part'" class="c-gscan-workflow-name">
+                    <span>{{ checkForBranchingLineage(scope.node) ? scope.node.name || scope.node.id : createDescendentLabel(scope.node) }}</span>
```

Comment:
I think this test might need to use `this.isExpanded` so that the name collapsing is only performed when the node is collapsed.

```
<span>{{ (this.isExpanded && checkForBranchingLineage(scope.node)) ? scope.node.name || scope.node.id : createDescendentLabel(scope.node) }}</span>
```

E.G. the collapsed state should look like this:

```
b/b1/b11
```

The expanded state should look as it did before:

```
b/
  b1/
    b11/
```

Rather than:

```
b/b1/b1
  b1/b11
    b11/
```

Reply:
Good point!